src/test/java/net/fornwall/jelf/ElfParserTest.java

```diff
@@ -0,0 +1,171 @@
+package net.fornwall.jelf;
+
+import com.diffblue.deeptestutils.Reflector;
+import com.diffblue.deeptestutils.mock.DTUMemberMatcher;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import java.io.ByteArrayInputStream;
+import java.lang.reflect.Method;
+
+@RunWith(PowerMockRunner.class)
+public class ElfParserTest {
+
+    @Test
+    public void testByteSwap() throws Exception {
+        ElfFile elfFile = (ElfFile)
+                Reflector.getInstance("net.fornwall.jelf.ElfFile");
+        ByteArrayInputStream fsFile = (ByteArrayInputStream)
+                Reflector.getInstance("java.io.ByteArrayInputStream");
+
+        ElfParser elfParser = new ElfParser(elfFile, fsFile);
+        Assert.assertEquals(16777216, elfParser.byteSwap(1));
+        Assert.assertEquals((short) 2560, elfParser.byteSwap((short) 10));
+        Assert.assertEquals(7205759403792793600L, elfParser.byteSwap(100L));
+    }
+
+    @Test
+    public void testReadUnsignedByte() throws Exception {
+        ElfParser elfParser = (ElfParser)
+                Reflector.getInstance("net.fornwall.jelf.ElfParser");
+
+        ByteArrayInputStream byteArrayInputStream = (ByteArrayInputStream)
+                Reflector.getInstance("java.io.ByteArrayInputStream");
+        Reflector.setField(byteArrayInputStream, "count", 1);
+        Reflector.setField(byteArrayInputStream, "buf", new byte[]{1});
+        Reflector.setField(elfParser, "fsFile", byteArrayInputStream);
+
+        Assert.assertEquals(1, elfParser.readUnsignedByte());
+    }
+
+    @Test
+    public void testReadShort() throws Exception {
+        ElfParser elfParser = (ElfParser)
+                Reflector.getInstance("net.fornwall.jelf.ElfParser");
+        ElfFile elfFile = (ElfFile)
+                Reflector.getInstance("net.fornwall.jelf.ElfFile");
+
+        ByteArrayInputStream byteArrayInputStream = (ByteArrayInputStream)
+                Reflector.getInstance("java.io.ByteArrayInputStream");
+        Reflector.setField(byteArrayInputStream, "count", 2);
+        Reflector.setField(byteArrayInputStream, "buf", new byte[]{1, 2});
```

Comment:
Every time a `ByteArrayInputStream` is created (e.g. lines `51-54` & others), you can do `new ByteArrayInputStream(new byte[]{1, 2, ...whatever})` instead of the reflective creation. I'm making an enhancement ticket for this.

Reply:
Done